### DISCUSSION TITLE
Concatenating two .tiff images results in a huge output file size

### DISCUSSION THREAD

**Discussion by ylieder** [ImageMagick/ImageMagick#2399] [Help] [answered]
I want two concatenate two .tiff images horizontally or vertically.
Given two images `a.tiff` with a size of 770x2455 and a file size of 25 KB and `b.tiff` with a size of 741x1489 and a file size of 20 KB.

Concatenating these images either vertically or horizontally should yield to an output file size of ~50 KB (in my imagination).

First try (horizontally):
`convert +append a.tiff b.tiff result.tiff` → Output file looks as expected, file size **469 KB**

Second try (vertically):
`convert -append a.tiff b.tiff result.tiff` → Output file looks as expected, file size **387 KB**

Third try (vertically):
`montage -mode concatenate -tile 1x a.tiff by.tiff result.tiff` → Output file looks as expected, file size **6,1 MB**

Is there a way to simply concatenate two .tiff images vertically/horizontally without scaling the output file size by 8-10?

**Comment by snibgo** [comment, marked as answer]
What version of IM, on what platform?

I suppose your input files are compressed. "convert a.tiff -verbose info:" and look for a line with "compression". Compression methods "LZW" and "ZIP" are lossless, and "JPEG" is lossy.

You can set an output compression method with a "-compress" option.

**Reply by ylieder** [reply]
Applied compression of input files was `group4`. 
The command `convert +append a.tiff b.tiff -compress group4 result.tiff` did the trick (36 KB output).

**Comment by snibgo** [comment]
Good, but you should put "+append" after reading the inputs, not before. Your command will work in old versions of IM, bt not in V7.

**Reply by ylieder** [reply]
Alright, good to know. 
But I'm on version 7.0.10-25 and the command still works 🤔

**Comment by snibgo** [comment]
You are using "convert", which give v6 syntax.

When you have v7, I suggest using "magick" instead. This provides more features, but is stricter about syntax.

**Reply by ylieder** [reply]
Okay thanks! 🙏
